src/App.tsx

```diff
@@ -17,18 +59,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setFilters={setHandleFilter}
+                setFilterInput={setFilterInput}
+                filterInput={filterInput}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              <TodoList
+                todos={handleInput()}
```

Comment:
so, if you apply `useMemo` hook you won't need to invoke function here yet just pass filtered todos